gridfs/asynchronous/grid_file.py

```diff
@@ -1194,19 +1194,9 @@ def __setattr__(self, name: str, value: Any) -> None:
                 )
```

Comment:
Rather than having `__setattr__` always raise an error in async, we should allow it as long as the file is not closed and only raise an error if the file is closed:
```python
                # All other attributes are part of the document in db.fs.files.
                # Store them to be sent to server on close() or if closed, send
                # them now.
                self._file[name] = value
                if self._closed:        
                    if _IS_SYNC:
                        self._coll.files.update_one({"_id": self._file["_id"]}, {"$set": {name: value}})
                    else:
                        raise AttributeError(
                            "AsyncGridIn does not support __setattr__ after being closed(). Set the attribute before closing the file or use AsyncGridIn.set() instead")
```